classes/plugins/PKPPubIdPluginHelper.inc.php

```diff
@@ -162,15 +162,20 @@ function execute($contextId, $form, $pubObject) {
 	 * @param $contextId integer
 	 * @param $form object
 	 * @param $pubObject object
+	 * @param $save boolean Whether the pub id shall be saved here
 	 * 	Submission, Representation, SubmissionFile + OJS Issue
 	 */
-	function assignPubId($contextId, $form, $pubObject) {
+	function assignPubId($contextId, $form, $pubObject, $save = false) {
 		$pubIdPlugins = PluginRegistry::loadCategory('pubIds', true, $contextId);
 		if (is_array($pubIdPlugins)) {
 			foreach ($pubIdPlugins as $pubIdPlugin) {
 				if ($form->getData($pubIdPlugin->getAssignFormFieldName())) {
 					$pubId = $pubIdPlugin->getPubId($pubObject);
-					$pubObject->setStoredPubId($pubIdPlugin->getPubIdType(), $pubId);
+					if ($save) {
```

Comment:
There isn't a PR for OMP, correct? Since the default for `$save` is false, won't this result in the pub ID not being saved in OMP anymore?